pkg/cmd/operator/mustgather.go

```diff
@@ -46,24 +46,40 @@ var (
 type MustGatherOptions struct {
 	*GatherBaseOptions
 
-	AllResources bool
+	AllResources            bool
+	CollectedResourceGroups []ResourceGroupSpec
 }
 
-func NewMustGatherOptions(streams genericclioptions.IOStreams) *MustGatherOptions {
-	return &MustGatherOptions{
-		GatherBaseOptions: NewGatherBaseOptions("scylla-operator-must-gather", true),
-		AllResources:      false,
+type MustGatherOptionsOverride func(options *MustGatherOptions)
+
+func WithCollectedResourceGroups(groups []ResourceGroupSpec) func(options *MustGatherOptions) {
+	return func(options *MustGatherOptions) {
+		options.CollectedResourceGroups = groups
 	}
 }
 
+func NewMustGatherOptions(streams genericclioptions.IOStreams, overrides ...MustGatherOptionsOverride) *MustGatherOptions {
```

Comment:
it's not required, there's a default value if not overwritten.

Reply:
why do you need the lambdas, when it returns the same, public type and you can set it directly without extra obfuscation? this pattern is meant when it returns an inteface and the caller can't set it directly or something within the cunstructor depends on this

Reply:
This pattern is also used for better UX allowing object initialization within single line. This is why I used it.

Reply:
IMO the UX got a lot worse since instead of an assignment you need to create a function + do the assignment. and this is only useful for private types and interfaces, not public ones.

Reply:
> IMO the UX got a lot worse

How it can get "a lot worse" when places where it was used doesn't require any changes? 

>  instead of an assignment you need to create a function + do the assignment. 

You still have a way to change these via assignment, or you can pass provided lambda, or write your own. 

IMO adding required parameter would make it worse, as something that can be defaulted would have to be passed explicitly.

Reply:
So what's the point of all of this extra code (that needs cross repo PRs to be adjusted), if you can just do
```golang
o := NewMustGatherOptions(streams)
o.CollectedResourceGroups = []string{o.CollectedResourceGroups, extraRG}
```
what I see is that we have more code to maintain so we can do equivalent function syntax that grows for every field like
```golang
type MustGatherOptionsOverride func(options *MustGatherOptions)

func WithCollectedResourceGroups(groups []GroupResourceSpec) func(options *MustGatherOptions) {
	return func(options *MustGatherOptions) {
		options.CollectedResourceGroups = groups
	}
}

o := NewMustGatherOptions(streams)
o.WithCollectedResourceGroups([]string{o.CollectedResourceGroups, extraRG})
```
and have extra function setters around the `NewMustGatherOptions` code?

That's 400% overhead (and I am not counting the test code for the unused function)

UX wise (caller side) looking up a function `WithCollectedResourceGroups` is an extra indirect step compared to just assigning the field directly. As the purpose isn't encapsulation of an interface, these extra setter aren't needed.

Reply:
Maybe if you would use it once, you would change your mind. Your example doesn't compile and argument about overhead is wrong because of it. 

Removed it anyway to avoid pointless discussion about personal preference.